src/diffusers/models/activations.py

```diff
@@ -98,9 +102,12 @@ def forward(self, hidden_states, *args, **kwargs):
         if len(args) > 0 or kwargs.get("scale", None) is not None:
             deprecation_message = "The `scale` argument is deprecated and will be ignored. Please remove it, as passing it will raise an error in the future. `scale` should directly be passed while calling the underlying pipeline component i.e., via `cross_attention_kwargs`."
             deprecate("scale", "1.0.0", deprecation_message)
-
-        hidden_states, gate = self.proj(hidden_states).chunk(2, dim=-1)
-        return hidden_states * self.gelu(gate)
+        if is_torch_npu_available():
+            hidden_states = self.proj(hidden_states, *args)
+            return torch_npu.npu_geglu(hidden_states, dim=-1, approximate=1)[0]
```

Comment:
Can we not use the existing `self.gelu()` when using NPU?

Reply:
Compared to self.gelu(), using torch_npu.npu_geglu can run faster and save memory on NPU.